sale_delivery_client_order_ref/models/stock_picking.py

```diff
@@ -0,0 +1,13 @@
+# Copyright 2019 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    client_order_ref = fields.Char(
+        related='group_id.sale_id.client_order_ref',
+        string='Customer Reference',
+    )
```

Comment:
I was thinking of the approach of extending `_prepare_procurement_values` instead of making the field a computed field.
https://github.com/odoo/odoo/blob/01caf1e44f653a8e9bf5edd583bc2421905d9464/addons/sale_stock/models/sale_order.py#L328-L352

But with this approach, I think you need to add the field in stock.move also if I'm not mistaken.  The approach of computed field might be better after all.